### ISSUE TITLE
Add lens, traversals and indexed traversals

### ISSUE THREAD

**Issue by infinity0** [dmwit/ordered-containers#20]
It would be nice to have lens instances so people using lens with Data.Map etc can easily migrate to this package.

**Comment by dmwit** [comment]
I'm open to this, but I don't have enough `lens` experience to know which instances are the right ones to write. Feel free to send me a pull request, though!

There's also the concern that `lens` is a pretty heavyweight dependency for what is otherwise a pretty light package. We could perhaps have separate but canonical packages with orphan instances, similar to how the `universe-*` family of packages work?

**Comment by infinity0** [comment]
Yes having orphan instances in a separate package such as `ordered-containers-lens` is a pretty common pattern for lens. As for writing them yourself, there are examples for the standard containers `Data.Map` etc in the `lens` package, you could hopefully just re-use them with minimal adjustments.
